src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -74,7 +74,16 @@ class MarathonScheduler(
     logger.info(s"Received status update for task ${status.getTaskId.getValue}: " +
       s"${status.getState} (${status.getMessage}, healthy: ${status.getHealthy})")
 
-    taskStatusProcessor.publish(status).failed.foreach {
+    // Mesos may produce a status update with really long messages on some cases. We need to restrict
+    // the message length to something we can actually store in a ZK node
+    val newStatus = if (status.hasMessage && status.getMessage.length > MarathonScheduler.MAX_STATUS_MESSAGE_LENGTH) {
+      logger.warn(s"Status update had length of ${status.getMessage.length}, longer than allowed ${MarathonScheduler.MAX_STATUS_MESSAGE_LENGTH} characters. Using only substring")
+      status.toBuilder.setMessage(status.getMessage.substring(0, MarathonScheduler.MAX_STATUS_MESSAGE_LENGTH)).build()
```

Comment:
I'd actually print the last line or X characters as well since these often include important details. Also, what do you think about printing the message?

Reply:
The message is printed in full anyway in the line before the truncation check. I think including the last X characters would make this not much cleaner, it should only be a safeguard anyway.